cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -361,8 +373,25 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 			string(common.AlreadyPresent),
 			string(common.MessageResourceAlreadyPresent),
 		)
-		// If the volume already present then return the cvr status as online
-		return string(apis.CVRStatusOnline), nil
+		// After creating zfs datasets in zpool but update to etcd might be
+		// failed
+		if isEmptyReplicaID(cVR) {
+			cVR.Spec.ReplicaID, err = volumereplica.GetReplicaIDFromZFS(fullVolName)
+			if err != nil {
+				// If error happened then update with same as with existing CVR
+				// phase. So, in next reconciliation it will try to update with
+				// proper changes
+				return string(cVR.Status.Phase), pkg_errors.Wrapf(err,
+					"volume replica %s exists", cVR.Name)
+			}
+		}
+		// If the volume already present then get the status of replica from ZFS
+		// and update it with corresponding status
+		status, err = volumereplica.Status(fullVolName)
+		if err != nil {
+			return string(cVR.Status.Phase), err
+		}
+		return status, nil
```

Comment:
can we write a func as we are using at number of place to retrun the status ?

Reply:
I will update it